src/boot_info.rs

```diff
@@ -8,7 +8,7 @@ use core::{ops, slice};
 /// ```
 /// # use bootloader::BootInfo;
 /// # type _SIGNATURE =
-/// extern "C" fn(boot_info: &'static BootInfo) -> !;
+/// fn(boot_info: &'static mut BootInfo) -> !;
```

Comment:
The `extern "C"` is still part of the signature, even if it's not needed with the `entry_point` macro. So I think we should keep it here and instead add to the docs of `entry_point!` that it is not needed when using the macro.

Reply:
You are totally correct, thanks!

The docs of the macro don't mention `extern "C"`, so I think it's fine. :thinking: